lib/src/object_id.rs

```diff
@@ -21,8 +21,25 @@ pub trait ObjectId {
     fn hex(&self) -> String;
 }
 
+// Defines a new struct type with visibility `vis` and name `ident` containing
+// a single Vec<u8> used to store an identifier (typically the output of a hash
+// function) as bytes. Types defined using this macro automatically implement
+// the `ObjectId` and `ContentHash` traits.
+// Documentation comments written inside the macro definition and will be
+// captured and associated with the type defined by the macro.
+//
+// Example:
+// ```no_run
+// id_type!(
+//     /// My favorite id type.
+//     pub MyId
+// );
+// ```
 macro_rules! id_type {
-    ($vis:vis $name:ident) => {
+    (   $(#[$($docs:meta)*])*
```

Comment:
perhaps, the inner repetition isn't needed?

`$(#[$attr:meta])*`

Reply:
Yeah, I think you're right. Maybe it's possible to match the individual items within the attribute block, but we don't really need to do that here.

I also renamed `docs` to `attr` since technically this matches any attributes, not just doc comments. Thanks!